### PR TITLE
ockam hub

### PR DESCRIPTION
**Pull request by mrinalwadhwa**

### CODE REVIEW

File: implementations/elixir/ockam/ockam/lib/ockam/secure_channel/channel.ex [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+    metadata = Map.put(metadata, :return_value, return_value)
+    Telemetry.emit_stop_event([__MODULE__, :handle_event], start_time, metadata: metadata)
+
+    return_value
+  end
+
+  defp handle_message(:info, event, {:key_establishment, _role, _role_state} = state, data) do
+    key_establisment_protocol = Map.get(data, :key_establisment_protocol)
+    key_establisment_protocol.handle_message(event, state, data)
```
Comment (nilslice) [inline implementations/elixir/ockam/ockam/lib/ockam/secure_channel/channel.ex:75]: typo: `s/establisment/establishment`

Comment (mrinalwadhwa) [reply implementations/elixir/ockam/ockam/lib/ockam/secure_channel/channel.ex:75]: thanks, will fix

File: implementations/elixir/ockam/ockam/lib/ockam/secure_channel/channel.ex [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+  # sets a key establishment protocol and calls its setup
+  defp setup_key_establisment_protocol(options, data) do
+    case Keyword.get(options, :key_establisment_protocol, XXKeyEstablishmentProtocol) do
+      XXKeyEstablishmentProtocol ->
+        data = Map.put(data, :key_establisment_protocol, XXKeyEstablishmentProtocol)
+        data.key_establisment_protocol.setup(options, data)
+
+      unexpected_protocol ->
+        {:error, {:unexpected_key_establisment_protocol, unexpected_protocol}}
```
Comment (nilslice) [inline implementations/elixir/ockam/ockam/lib/ockam/secure_channel/channel.ex:122]: same typo: `s/establisment/establishment/c`

File: implementations/elixir/ockam/ockam/lib/ockam/secure_channel/channel.ex (hunk 1 above)
Comment (mrinalwadhwa) [reply implementations/elixir/ockam/ockam/lib/ockam/secure_channel/channel.ex:75]: ignore any repetitions of this ... I'll search and replace

**Comment by nilslice** [comment]
Looks good, @mrinalwadhwa -- I'll await new commit for typo fixes before approving is it will be auto-dismissed.

**Review by nilslice — APPROVED** [review]

**Pull request merged**